perun-registrar-lib/src/main/java/cz/metacentrum/perun/registrar/modules/Vsup.java

```diff
@@ -40,55 +43,87 @@
 	private final static Logger log = LoggerFactory.getLogger(Vsup.class);
 
 	@Override
-	public List<ApplicationFormItemData> createApplication(PerunSession session, Application application, List<ApplicationFormItemData> data) throws PerunException {
+	public void canBeApproved(PerunSession session, Application app) throws PerunException {
 
-		if (application.getUser() == null) {
+		List<ApplicationFormItemData> data = registrar.getApplicationDataById(session, app.getId());
+
+		if (app.getUser() == null) {
+
+			for (ApplicationFormItemData item : data) {
+				if (item.getFormItem() != null &&
+					Objects.equals(AttributesManager.NS_USER_ATTR_DEF + ":birthNumber", item.getFormItem().getPerunDestinationAttribute())) {
+
+					// if application contains birth number, try to map to existing user
+					String rc = item.getValue();
+					if (rc != null && !rc.isEmpty()) {
+
+						try {
+							User user = ((PerunBl) session.getPerun()).getUsersManagerBl().getUserByExtSourceNameAndExtLogin(session, "RC", rc);
+							throw new CantBeApprovedException("Application has the same birth number " + rc + " as user " + user + " already in Perun and thus would be merged with him.", null, null, null, true);
```

Comment:
Message will contain toString() from the User object, which is not most user friendly way. I would prefer printing just displayName and ID.

Reply:
Fixed.